jlm/llvm/ir/linkage.hpp

```diff
@@ -31,6 +34,26 @@ is_externally_visible(const linkage & lnk)
   return lnk != linkage::internal_linkage;
 }
 
+static inline std::string
+linkage_to_string(const linkage & lnk)
+{
+  std::unordered_map<linkage, std::string> strings = {
```

Comment:
It should be `static std::unordered_map<...>`, otherwise the map is created every time this function is invoked and not just once.

Reply:
I get the following error when removing inline that I don't understand how to resolve.

```
g++ -c --std=c++17 -Wall -Wpedantic -Wextra -Wno-unused-parameter -Werror -Wfatal-errors -gdwarf-4 -g -O3 -Wno-error=comment -I. -Itests -I/usr/lib/llvm-16/include  -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -I./build-circt/circt//include -I./build-mlir/mlir//include -o build/jlm/llvm/frontend/ControlFlowRestructuring.la jlm/llvm/frontend/ControlFlowRestructuring.cpp
In file included from ./jlm/llvm/ir/variable.hpp:9,
                 from ./jlm/llvm/ir/cfg.hpp:12,
                 from ./jlm/llvm/ir/basic-block.hpp:10,
                 from jlm/llvm/frontend/ControlFlowRestructuring.cpp:6:
./jlm/llvm/ir/linkage.hpp:38:1: error: ‘std::string jlm::llvm::linkage_to_string(const jlm::llvm::linkage&)’ defined but not used [-Werror=unused-function]
   38 | linkage_to_string(const linkage & lnk)
      | ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
```

Reply:
I fail to understand what this error message has to do with putting a `static` in front of the `std::unordered_map<>()`. Your error message looks more like you removed the `inline` from before the linkage_to_string() method.

Reply:
I missunderstod your previous comment.
Should be fixed now.